tests/parallel/parallel_test.C

```diff
@@ -151,18 +151,16 @@ public:
 
   void testBroadcast()
   {
-    std::vector<unsigned int> src(3), dest(3);
-
-    src[0]=0;
-    src[1]=1;
-    src[2]=2;
+    // Workaround for spurious warning from operator=
+    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100366
+    std::vector<unsigned int> src{0,1,2}, dest(3,0);
```

Comment:
I'm always amazed by what C++ compiler writers do. Sometimes in a good way, sometimes not so good.

Reply:
Yeah, I was shocked by this bug and utterly horrified by the optimization-corruption bug, but I must admit that the reason for my shock and horror is that compiler writers have given me extremely high expectations over the years.  If I tried to write a modern C++ compiler I'm sure it would fail to compile half of the valid codes out there and the ones that compiled would run ten times slower, yet the last time I can recall being affected by a serious compiler bug was before gcc 3 came out.